nautobot/ipam/querysets.py

```diff
@@ -305,7 +305,7 @@ def delete(self, *args, **kwargs):
             return super().delete(*args, **kwargs)
         except ProtectedError as err:
             # This will be either IPAddress or Prefix.
-            protected_instance = tuple(err.protected_objects)[0]
```

Comment:
While the justification for this [rule](https://docs.astral.sh/ruff/rules/unnecessary-iterable-allocation-for-first-element/#unnecessary-iterable-allocation-for-first-element-ruf015) makes sense I don't like how difficult it is to read this line. Maybe eventually my brain will learn to associate `next(iter(list))` with grabbing the first element of `list`..

Reply:
Agreed - this is one I considered disabling when I ran across it, because the replacement syntax is definitely non-obvious.